addons/sale_pdf_quote_builder/static/src/js/quotation_document_kanban/quotation_document_kanban_widget.js

```diff
@@ -18,9 +19,11 @@ export class QuotationDocumentX2ManyField extends X2ManyField {
     setup() {
         super.setup();
         this.uploadRoute = '/sale_pdf_quote_builder/quotation_document/upload';
-        this.formData = {
-            'sale_order_template_id': this.props.record.resId,
-        };
+        onWillRender(() => {
+            this.formData = {
+                'sale_order_template_id': this.props.record.resId,
```

Comment:
```suggestion
                'sale_order_template_id': this.props.record.resId || 0,
```
This would solve the `ValueError` in Python, and allow uploading documents before save.